### PR TITLE
Target sdk aliases and exclusions

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @gribozavr

File: cmake/modules/SwiftUtils.cmake [hunk 1]
```diff
@@ -3,3 +3,12 @@ function(precondition var)
     message(FATAL_ERROR "Error! Variable ${var} is false or not set!")
   endif()
 endfunction()
+
+macro(list_replace lst old new)
```
Comment (gribozavr) [inline cmake/modules/SwiftUtils.cmake]: We are trying really hard to use functions when possible.  Could you change this into a function?  You can use PARENT_SCOPE to set the variable in the caller.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@compnerd

```
ninja: error: '/home/buildnode/jenkins/workspace/swift-PR-Linux@2/swift/test/swiftStdlibCollectionUnittest-linux', needed by 'test/CMakeFiles/check-swift-all-linux-x86_64', missing and no known rule to make it
```

File: cmake/modules/SwiftUtils.cmake (hunk 1 above)
Comment (compnerd) [inline cmake/modules/SwiftUtils.cmake]: Yeah, I agree, functions are better as they don't pollute the global namespace.  Ill see if I can get that changed.

**Comment by compnerd** [comment]
@swift-ci Please test

**Comment by compnerd** [comment]
@gribozavr okay to merge?

**Pull request merged**